### DISCUSSION TITLE
How to pass a CUDA buffer into ONNX session ?

### DISCUSSION THREAD

**Discussion by boxerab** [microsoft/onnxruntime#16602] [General] [answered]
I am using CUDA EP with C++ API. Currently I pass system memory buffers to the session for inference.
I have another use case where the image is already stored in a CUDA buffer. Is there a way to pass this CUDA
buffer to ONNX without  downloading it to system memory? Thanks!

**Comment by baijumeswani** [comment, marked as answer]
You can do this with [`IOBinding`](https://onnxruntime.ai/docs/api/c/struct_ort_1_1detail_1_1_session_impl.html#aba18502e41f8dd46038e3de1efa363b5).

Example usage in Python:

https://onnxruntime.ai/docs/api/python/api_summary.html#data-on-device

**Reply by boxerab** [reply]
Great! thanks for your quick reply. I will use that.
